galley/cmd/galley/cmd/root.go

```diff
@@ -48,21 +63,96 @@ func GetRootCmd(args []string, printf, fatalf shared.FormatFn) *cobra.Command {
 			if len(args) > 0 {
 				return fmt.Errorf("%q is an invalid argument", args[0])
 			}
-
 			return log.Configure(loggingOptions)
 		},
+		Run: func(cmd *cobra.Command, args []string) {
+
+			sa.KubeConfig = flags.kubeConfig
+			sa.ResyncPeriod = flags.resyncPeriod
+			sa.CACertificateFile = vc.CACertFile
+			sa.KeyFile = vc.KeyFile
+			sa.CertificateFile = vc.CertFile
+			noServer := flags.noServer
+			noValidation := flags.noValidation
+
+			if noServer && noValidation {
+				fatalf("Galley must be running under at least one mode: server or validation")
+			}
+
+
+			if !noServer {
+				go server.RunServer(sa, printf, fatalf)
+			}
+			if !noValidation {
+				go validation.RunValidation(vc, printf, fatalf, flags.kubeConfig)
+			}
+
+			//Run liveness and readiness check
+			probeCheckController := galleyProbe.NewProbeCheckController(flags.livenessProbeOptions,flags.readinessProbeOptions)
+			probeCheckController.Run()
+			galleyStop := make(chan struct{})
+
+			istiocmd.WaitSignal(galleyStop)
+
+
+		},
+
 	}
+
 	rootCmd.SetArgs(args)
 	rootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
-
+	flags.livenessProbeOptions = &probe.Options{}
+	flags.readinessProbeOptions = &probe.Options{}
 	rootCmd.PersistentFlags().StringVar(&flags.kubeConfig, "kubeconfig", "",
 		"Use a Kubernetes configuration file instead of in-cluster configuration")
-
 	rootCmd.PersistentFlags().DurationVar(&flags.resyncPeriod, "resyncPeriod", 0,
 		"Resync period for rescanning Kubernetes resources")
+	rootCmd.PersistentFlags().BoolVar(&flags.noServer, "no-server", false,
+		"Run galley server mode")
+	rootCmd.PersistentFlags().BoolVar(&flags.noValidation, "no-validation", false,
+		"Run galley validation mode")
```

Comment:
Can we reverse the polarity of `no-server and `no-validation` to avoid double negatives? I think the command line usage would read a bit better, e.g.

```bash
galley --no-server=true --no-validation=false
// vs.
galley --enable-server=false --enable-validation
```

Reply:
Done~